packages/core/src/context/MemoryStore.example.ts

```
@@ -0,0 +1,352 @@
+    console.log('Reading memories from multiple files...');
+    const allMemories = await memoryStore.readAllMemories();
+    
+    // Group memories by file
+    const memoriesByFile = new Map<string, typeof allMemories>();
+    
+    allMemories.forEach(memory => {
+      const fileName = memory.filePath.split('/').pop() || 'unknown';
+      if (!memoriesByFile.has(fileName)) {
+        memoriesByFile.set(fileName, []);
+      }
+      memoriesByFile.get(fileName)!.push(memory);
```

Biome (1.9.4)

[error] Forbidden non-null assertion.

Unsafe fix: Replace with optional chain operator ?. This operator includes runtime checks, so it is safer than the compile-only non-null assertion operator


(lint/style/noNonNullAssertion)